### PR TITLE
Jenkinsfile: migration to shared library & cleanup

### PR DESCRIPTION
**Pull request by peat-psuwit**
This PR moves all code for Jenkinsfile to a shared library that can be included from all repositories. This makes sure we DRY and allows us to implement a fix that applies to all repositories (should there be a need in the future).

CI needs to include this repository in the library list to be able to utilize the shared library.

### CODE REVIEW

File: vars/buildAndProvideDebianPackage.groovy [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+              cleanWs(cleanWhenAborted: true, cleanWhenFailure: true, cleanWhenNotBuilt: true, cleanWhenSuccess: true, cleanWhenUnstable: true, deleteDirs: true)
+              unstash 'source'
+              sh 'architecture="armhf" build-binary.sh'
+              stash(includes: stashFileList, name: 'build-armhf')
+              cleanWs(cleanWhenAborted: true, cleanWhenFailure: true, cleanWhenNotBuilt: true, cleanWhenSuccess: true, cleanWhenUnstable: true, deleteDirs: true)
+            }
+          }
+          stage('Build binary - arm64') {
+            agent { label 'arm64' }
+            // Always run; arch-independent packages are built here.
```
Comment (mariogrip) [inline vars/buildAndProvideDebianPackage.groovy]: We might want to do arch independent packages on amd64 as its much faster

We can achieve this by setting
```
export architecture="all"
export FORCE_BINARY_ONLY="all"
```

Comment (peat-psuwit) [reply vars/buildAndProvideDebianPackage.groovy]: Maybe it's better to set `MAIN_ARCHITECTURE` correctly in `/etc/jenkins/debian_glue`? This way, packages with a mix of arch-dependent & arch-independent binary packages will benefit, too.

The condition & missing of it has to match the architecture that will build arch-independent packages.

**Comment by peat-psuwit** [comment]
A build using `buildAndProvideDebianPackage()`.
https://ci.ubports.com/blue/organizations/jenkins/ubports%2Fgst-droid-packaging/detail/xenial_-_gst-droid/7/pipeline

A build using `buildAndProvideDebianPackage(/* isArchIndependent */ true)` (seems like BlueOcean's live view is confused by `when` directive, but the build is fine it seems).
https://ci.ubports.com/blue/organizations/jenkins/ubports%2Fdroidmedia-packaging/detail/xenial_-_gst-droid/8

File: vars/buildAndProvideDebianPackage.groovy (hunk 1 above)
Comment (mariogrip) [reply vars/buildAndProvideDebianPackage.groovy]: This as been done, "all" is now built on amd64

**Review by mariogrip — APPROVED** [review]
LGTM :+1:

**Pull request merged**